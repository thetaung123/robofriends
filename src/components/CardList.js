import React from 'react';
import Card from "./Card";

const CardList = ({robots}) => { //calling props from app
    return (
          <div>
            {
                robots &&
                robots.map((users,i)=>{
                return (<Card
                key={robots[i].id}
                id={robots[i].id}
                name={robots[i].name}
                email={robots[i].email}
                />); //to avoid errors if one of the items in the array , we need to add key props whenever we loop an array
                })
            }
        </div>
    );
};          // find out more about it by deleting key props and reading the error in the console // key props should be a number that doesn't change like id

export default CardList;
