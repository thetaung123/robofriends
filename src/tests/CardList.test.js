import { shallow } from "enzyme";
import React from "react";
import CardList from "../components/CardList";
import { shallowToJson } from "enzyme-to-json"; // needed for new Jest version
// because we're using CRA, we don't need to install Jest // and we can run tests by entering npm test in the terminal
// we can use jest functions here
it('expect to render cardlist component', () => {
  const mockRobots = [
    {
      id: 1,
      name: "John Snow",
      username: "snowwhite",
      email: "john.s@gmail.com"
    }
  ]
  expect(shallowToJson(shallow(<CardList robots={mockRobots}/>))).toMatchSnapshot(); // this snapshot the component when the test first run, under __snapshots__ folder, so that if the component is changed the test will show error. // if you don't want that you can update the snapshot by pressing "u" key in the terminal while the test in watch mode
});
