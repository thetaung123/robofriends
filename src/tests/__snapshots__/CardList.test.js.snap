// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render cardlist component 1`] = `
<div>
  <Card
    email="john.s@gmail.com"
    id={1}
    key="1"
    name="John Snow"
  />
</div>
`;
